chain/traversal.go

```diff
@@ -87,3 +87,33 @@ func TipSetProviderFromBlocks(ctx context.Context, blocks BlockProvider) TipSetP
 func (p *tipsetFromBlockProvider) GetTipSet(tsKey types.TipSetKey) (types.TipSet, error) {
 	return LoadTipSetBlocks(p.ctx, p.blocks, tsKey)
 }
+
+// CollectTipsToCommonAncestor traverses chains from two tipsets (called old and new) until their common
+// ancestor, collecting all tipsets that are in one chain but not the other.
+// The resulting lists of tipsets are ordered by decreasing height.
+func CollectTipsToCommonAncestor(ctx context.Context, store TipSetProvider, oldHead, newHead types.TipSet) (oldTips, newTips []types.TipSet, err error) {
```

Comment:
Mixed feelings the named returns :thinking:, I suppose this does improve readability.

Reply:
Named returns are a good idea if you are returning multiples of the same type, that's my 2 centavos.